apiserver/facades/controller/caasunitprovisioner/provisioner.go

```diff
@@ -673,14 +677,14 @@ func (a *Facade) updateUnitsFromCloud(app Application, scale *int, unitUpdates [
 		return errors.Trace(err)
 	}
 
-	if scale == nil {
+	if scale == nil || generation == nil {
```

Comment:
this will break older operators
if generation == nil we need to default to 0 and continue

Reply:
ops, fixed! thx